grpc/ra-wrappers.go

```diff
@@ -121,15 +121,8 @@ func (rac RegistrationAuthorityClientWrapper) AdministrativelyRevokeCertificate(
 	return nil
 }
 
-func (ras *RegistrationAuthorityClientWrapper) NewOrder(ctx context.Context, request *rapb.NewOrderRequest) (*corepb.Order, error) {
-	resp, err := ras.inner.NewOrder(ctx, request)
-	if err != nil {
-		return nil, err
-	}
-	if resp == nil || !orderValid(resp) {
```

Comment:
I don't see any calling/client code (probably in the WFEs) checking the validity of the returned order. It would be easy to miss, because the signature of this method was already all-protobufs so the interface didn't need to be updated, but we should still take a look at all call locations and see what additional validation (if any) they want to do.

Reply:
It's there. It is on the same line as the error check after the call. I can separate if you prefer but it probably should return the same error so it seemed like the right place. 
https://github.com/letsencrypt/boulder/blob/c52fae0e94a657c699baa2a4f5057491ca31493b/wfe2/wfe.go#L2073